src/components/Tabs.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+
+type Props = {
+  tabs: Tab[];
+  onTabSelected: (tab: Tab) => void;
+  selectedTab: Tab;
+};
+
+export class Tabs extends React.PureComponent<Props, {}> {
+  render() {
+    const { tabs, onTabSelected, selectedTab } = this.props;
+
+    return (
+      <>
+        <ul className="list">
+          {[...tabs].map(tab => (
```

Comment:
map doesn't mutate array

```suggestion
          {tabs.map(tab => (
```